src/Integrations/Integrations/LaravelQueue/LaravelQueueIntegration.php

```diff
@@ -0,0 +1,333 @@
+            'add',
+            function (SpanData $span, $args, $retval, $exception) use ($integration) {
+                $integration->setSpanAttributes(
+                    $span,
+                    'laravel.queue.batch.add',
+                    'send',
+                    null,
+                    $exception,
+                    null,
+                    get_class($this)
+                );
+                if ($retval) {
+                    $span->meta['messaging.laravel.batch_id'] = $retval->id;
+                }
+            }
+        );
+
+        return Integration::LOADED;
+    }
+
+    public function setSpanAttributes(
+        SpanData $span,
+        string $name,
+        $operation = null,
+        $job = null,
+        $exception = null,
+        $queue = null,
+        $resourceSubstitute = null
+    ) {
+        $span->name = $name;
+        $span->service = $this->getAppName();
+        $span->type = 'queue';
+        $span->meta[Tag::SPAN_KIND] = 'client';
+        $span->meta[Tag::COMPONENT] = LaravelQueueIntegration::NAME;
+
+        if ($operation) {
+            $span->meta[Tag::MQ_OPERATION] = $operation;
+        }
+
+        if ($job instanceof Job) {
+            $jobName = $job->resolveName();
+            $span->meta = array_merge(
+                $span->meta,
+                $this->getMetadataFromJob($job)
+            );
+            $queue = $queue ?? $job->getQueue();
+        } elseif (is_object($job)) { // Most certainly a CallQueuedClosure
+            $jobName = method_exists($job, 'displayName') ? $job->displayName() : get_class($job);
+            $connectionName = $job->connection ?? config('queue.default');
+            $queue = $queue ?? ($job->queue ?? (config("queue.connections.$connectionName.queue") ?? 'default'));
+
+            $span->meta['messaging.laravel.name'] = $jobName;
+            $span->meta['messaging.laravel.connection'] = $connectionName;
+            $span->meta[Tag::MQ_DESTINATION] = $queue;
+            $span->meta = array_merge(
+                $span->meta,
+                $this->getMetadataFromObject($job)
+            );
+        } else {
+            $jobName = $job;
+        }
+
+        if ($resourceSubstitute) {
+            $span->resource = $resourceSubstitute;
+        } else {
+            $span->resource = $queue ? "$jobName -> $queue" : "$jobName";
+        }
+
+        if ($exception) {
+            $this->setError($span, $exception);
+        }
+    }
+
+    public function getMetadataFromJob(Job $job)
+    {
+        $metadata = [
+            'messaging.laravel.attempts' => $job->attempts(),
+            'messaging.laravel.connection' => $job->getConnectionName() ?? config('queue.default'),
+            'messaging.laravel.max_tries' => $job->maxTries(),
+            'messaging.laravel.timeout' => $job->timeout(),
+            'messaging.laravel.name' => $job->resolveName(),
+            Tag::MQ_SYSTEM => 'laravel',
```

Comment:
Isn't this supposed to be like `kafka`, `rabbitmq`, `rocketmq`, `activemq`, `AmazonSQS`, etc?

Reply:
Well, Laravel queues aren't dispatched via these mechanisms. It's a separate mechanism.

Reply:
On this, I was also planning to update the confluence page to include the newly introduced span attributes; `messaging.system` would be one of these.